timelock-impl/src/main/java/com/palantir/atlasdb/timelock/AsyncTimelockServiceImpl.java

```diff
@@ -124,21 +119,42 @@ public StartAtlasDbTransactionResponse startAtlasDbTransaction(IdentifiedTimeLoc
     @Override
     public StartAtlasDbTransactionResponseV3 startIdentifiedAtlasDbTransaction(
             StartIdentifiedAtlasDbTransactionRequest request) {
+        Leased<LockImmutableTimestampResponse> leasedLockImmutableTimestampResponse =
+                lockImmutableTimestampWithLease(request.requestId());
+
+        TimestampAndPartition timestampAndPartition =
+                timestampService.getFreshTimestampForClient(request.requestorId());
+
+        return StartAtlasDbTransactionResponseV3.of(
+                leasedLockImmutableTimestampResponse.value(),
+                timestampAndPartition,
+                leasedLockImmutableTimestampResponse.lease());
+    }
+
+    @Override
+    public BatchedStartTransactionResponse batchedStartTransaction(BatchedStartTransactionRequest request) {
```

Comment:
in line with comments on the old PR - think this isn't a new 'batched' version of starting transaction, I think it's actually startAtlasDbTransactionV4

Reply:
(which might well lead to some class renames here)